torch/CMakeLists.txt

```diff
@@ -347,45 +337,76 @@ install(TARGETS torch
   LIBRARY DESTINATION "${TORCH_INSTALL_LIB_DIR}"
   ARCHIVE DESTINATION "${TORCH_INSTALL_LIB_DIR}")
 
-if (TORCH_BUILD_TEST)
-  # JIT Tests. TODO: Put into test/cpp/jit folder
-
+# JIT Tests. TODO: Put into test/cpp/jit folder
+if (NOT MSVC AND NOT APPLE)
   add_executable(test_jit ${TORCH_SRC_DIR}/csrc/jit/test_jit.cpp)
+  target_link_libraries(test_jit torch ${TORCH_CUDA_LIBRARIES})
+  target_compile_definitions(test_jit PUBLIC USE_CATCH _FORCE_INLINES)
+  target_include_directories(test_jit PUBLIC
+    "${TORCH_SRC_DIR}/../third_party/catch/single_include"
+    ${ATen_CPU_INCLUDE})
 
-  target_link_libraries(test_jit torch)
+  if (USE_CUDA)
+    target_link_libraries(test_jit ${CUDA_LIBRARIES})
+  endif()
+endif()
 
-  target_include_directories(test_jit PUBLIC
-    "${TORCH_SRC_DIR}/../third_party/catch/single_include")
-
-  # API Tests
-
-  if (NOT NO_API)
-    set(TORCH_API_TEST_DIR "${TORCH_SRC_DIR}/../test/cpp/api")
-
-    add_executable(test_api
-      ${TORCH_API_TEST_DIR}/any.cpp
-      ${TORCH_API_TEST_DIR}/modules.cpp
-      ${TORCH_API_TEST_DIR}/cursor.cpp
-      ${TORCH_API_TEST_DIR}/integration.cpp
-      ${TORCH_API_TEST_DIR}/main.cpp
-      ${TORCH_API_TEST_DIR}/misc.cpp
-      ${TORCH_API_TEST_DIR}/module.cpp
-      ${TORCH_API_TEST_DIR}/optim.cpp
-      ${TORCH_API_TEST_DIR}/sequential.cpp
-      ${TORCH_API_TEST_DIR}/rnn.cpp
-      ${TORCH_API_TEST_DIR}/serialization.cpp
-      ${TORCH_API_TEST_DIR}/static.cpp
-      ${TORCH_API_TEST_DIR}/tensor.cpp
-      ${TORCH_API_TEST_DIR}/tensor_cuda.cpp
-      # Temporary until ATen tests are built with Caffe2
-      ${TORCH_API_TEST_DIR}/tensor_options.cpp
-      ${TORCH_API_TEST_DIR}/tensor_options_cuda.cpp
-    )
+if (NOT NO_API AND NOT USE_ROCM)
+  target_include_directories(torch PUBLIC
+    ${TORCH_SRC_DIR}/csrc/api
+    ${TORCH_SRC_DIR}/csrc/api/include)
 
-      target_include_directories(test_api
-        PUBLIC
-        "${TORCH_SRC_DIR}/../third_party/catch/single_include")
+  if (NOT MSVC)
+    target_compile_options(torch PRIVATE -Wno-error=maybe-uninitialized)
+  endif()
+
+  if (APPLE)
+    target_compile_options(torch PRIVATE -Wno-error=unknown-warning-option)
+  endif()
+endif()
 
-    target_link_libraries(test_api torch)
+if (NOT NO_API AND NOT USE_ROCM)
```

Comment:
Why not absorb this if block into the one that's immediately above it?

Reply:
++